File: Objective-C/ToDoList/AddRabbitaskVC.m
//
//  AddRabbitaskVC.m
//
//
//  Created by Eric Giannini on 5/21/15.
//
//

#import "AddRabbitaskVC.h"

@interface AddRabbitaskVC ()

@property (weak, nonatomic) IBOutlet UITextField *textField;
@property (weak, nonatomic) IBOutlet UIBarButtonItem *saveButton;


@end

@implementation AddRabbitaskVC

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view from its nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


#pragma mark - Navigation

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    if (sender != self.saveButton) return;
    if (self.textField.text.length > 0) {
        self.rabbitask = [[Rabbitask alloc] init];
        self.rabbitask.itemName = self.textField.text;
        self.rabbitask.completed = NO;
    }
}
@end

File: Swift/Rabbitask/AddRabbitaskVC.swift
//
//  AddRabbitaskVC.swift
//  Rabbitask
//
//  Created by Eric Giannini on 3/3/17.
//  Copyright © 2017 Unicorn Mobile, LLC. All rights reserved.
//

import UIKit

class AddRabbitaskVC : UIViewController {
    
    
    @IBOutlet weak var textField: UITextField!
    let toDoRabbitask = Rabbitask()
    
    
    @IBOutlet weak var saveButton: UIBarButtonItem!
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        
    }
    
    // MARK: - Navigation
    
    // In a storyboard-based application, you will often want to do a little preparation before navigation
    
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {

        // FIXME: empty cell returns even if textField is empty
        
        if (sender as? UIBarButtonItem) !== saveButton {
            return
        }
        if self.textField.text?.isEmpty == false {
            self.toDoRabbitask.itemName = self.textField.text!
        }
        
        
        
    }
    
}


